internal/resourcepolicies/resource_policies.go

```diff
@@ -69,46 +74,69 @@ func (policies *Policies) buildPolicy(resPolicies *resourcePolicies) error {
 		p.conditions = append(p.conditions, &storageClassCondition{storageClass: con.StorageClass})
 		p.conditions = append(p.conditions, &nfsCondition{nfs: con.NFS})
 		p.conditions = append(p.conditions, &csiCondition{csi: con.CSI})
-		policies.VolumePolicies = append(policies.VolumePolicies, p)
+		policies.volumePolicies = append(policies.volumePolicies, p)
 	}
 
 	// Other resource policies
 
-	policies.Version = resPolicies.Version
+	policies.version = resPolicies.Version
 	return nil
 }
 
-func (p *Policies) Match(res interface{}) *Action {
-	vol, ok := res.(*StructuredVolume)
-	if ok {
-		for _, policy := range p.VolumePolicies {
-			isAllMatch := false
-			for _, con := range policy.conditions {
-				if !con.Match(vol) {
-					isAllMatch = false
-					break
-				}
-				isAllMatch = true
-			}
-			if isAllMatch {
-				return &policy.action
+func (p *Policies) match(res *structuredVolume) *Action {
+	for _, policy := range p.volumePolicies {
+		isAllMatch := false
+		for _, con := range policy.conditions {
+			if !con.match(res) {
+				isAllMatch = false
+				break
 			}
+			isAllMatch = true
+		}
+		if isAllMatch {
+			return &policy.action
 		}
 	}
 	return nil
 }
 
+func (p *Policies) GetMatchAction(res interface{}, resType ResourceType) (*Action, error) {
```

Comment:
You can get the type of the interface by type assertion and don't need to pass the `resType` explicitly

Reply:
ok, I've removed resType